app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -215,6 +215,12 @@ class HomeFragment : Fragment() {
             sessionControlInteractor,
             homeViewModel
         )
+
+        // This has to be called separately from the consumeFrom block since the coroutine doesn't
+        // allow our UI to be updated right away and delays our start up. The init block allows us to
+        // force our UI to render with the data available in our store at fragment creation time.
+        sessionControlView?.update(homeFragmentStore.state)
```

Comment:
nit: move next to `consumeFrom` block you're referecing.

Also, it might be helpful to encapsulate them both in a helper function so the grouping is clear:
```kotlin
fun updateSessionControlView() { // maybe a better name
    fun update(state) {
        sessionControlView?.update(state)
    }

    // explain update now
    update(homeFragmentStore.state)
    
    // explain update later
    addObserver(::update)
}
```

Two best practices in play here:
- when I'm adding code with comments, I always think if it'd be more appropriate to put the code in a function to encapsulate the complexity
- when I'm adding code to lifecycle methods, I always think about using a helper to keep the lifecycle methods short and discussing things at a high level (this doesn't seem to be common on Fenix but I think it greatly helps clarity)